lesson14/main.js

```diff
@@ -71,20 +71,33 @@ const init = async () => {
   createNewList(val);
 };
 
-openModal.addEventListener("click", () => {
+const getNumbers = () => {
+  const inputNumber = document.getElementById("js-number");
+  return inputNumber.value;
+}
+
+const showModal = () => {
   modal.classList.remove("hidden");
   mask.classList.remove("hidden");
-});
+}
 
-closeModal.addEventListener("click", () => {
+const hiddenModal = () => {
   modal.classList.add("hidden");
   mask.classList.add("hidden");
+}
+
+openModal.addEventListener("click", () => {
+  showModal();
+});
+
+closeModal.addEventListener("click", () => {
+  hiddenModal();
 });
 
 requestBtn.addEventListener("click", () => {
+  console.log(getNumbers());
   init();
```

Comment:
IMO
I thought it would be better to use the init function to get the value.
> クリックした際にinput(type number)のvalueを取得して
> Promiseを実行する手前でconsole.log出力されてればいいです

This is what the problem says, but with the current code, it is displayed the moment the button is pressed.
```suggestion
requestBtn.addEventListener("click", () => {
  console.log(getNumbers());
  init(getNumbers());
```
init function

```js
const init = async (inputNumber) => {
  console.log(inputNumber); 　//Here is the console.log☺️
  createLoader();
  loading();
  const val = await fetchedData();
  createNewList(val);
};
```